src/logger/POSTRequestLogger.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Logger, LogLevel } from '../index';
+import IntervalScheduler from '../scheduler/IntervalScheduler';
+
+export default class POSTRequestLogger implements Logger {
+  private logCapture: string[] = [];
+  private logSequenceNumber: number = 0;
+  private lock = false;
+  private static BATCH_SIZES = 75;
+  private intervalScheduler: IntervalScheduler;
+
+  constructor(
+    private name: string,
+    private meetingId: string,
+    private attendeeId: string,
+    private intervalMs: number,
+    private baseURL: string,
+    private level = LogLevel.WARN,
+  ) {
+    this.intervalScheduler = new IntervalScheduler(this.intervalMs);
+    var _this = this;
+    window.addEventListener('unload', function(event) {
+      _this.intervalScheduler.stop();
+      if(!navigator.sendBeacon){
+        return
+      }
+      let body = JSON.stringify({
+        meetingId: _this.meetingId,
+        attendeeId: _this.attendeeId,
+        appName: _this.name,
+        logs: _this.logCapture
+      });
+      navigator.sendBeacon(_this.baseURL, body);
+    });
+  }
+  debug(debugFunction: () => string): void {
+    if (LogLevel.DEBUG < this.level) {
+      return;
+    }
+    this.log(LogLevel.DEBUG, debugFunction());
+  }
+
+  info(msg: string): void {
+    this.log(LogLevel.INFO, msg);
+  }
+
+  warn(msg: string): void {
+    this.log(LogLevel.WARN, msg);
+  }
+
+  error(msg: string): void {
+    this.log(LogLevel.ERROR, msg);
+  }
+
+  setLogLevel(level: LogLevel): void {
+    this.level = level;
+  }
+
+  getLogLevel(): LogLevel {
+    return this.level;
+  }
+
+  async publishLogsCaller(flushAll: boolean): Promise<void> {
+    if (flushAll === true && this.logCapture.length > 0) {
```

Comment:
This ```if``` block for flushAll is no more required since the beacon API sends all logs?